### ISSUE TITLE
Field filter with default value via query does not insert correct SQL for date filter widget

### ISSUE THREAD

**Issue by almegeddon** [metabase/metabase#9489]
### Environment details:

* Browser: Chrome Version 72.0.3626.119 (Official Build) (64-bit)
* Operating system: OS X 10.14.3
* Metabase: v0.31.2
* Metabase Hosting: Docker via AWS Fargate
* Metabase Internal DB: MySQL (via 
* Database: BigQuery

### Issue details/Steps to reproduce:

Setting a default value via query works great for a single date filter widget as prescribed by "Default value in the query" [section here](https://metabase.com/docs/latest/users-guide/13-sql-parameters.html).

When changing the widget to "Date Filter" (to allow for ranges, relative dates, etc.) the default value is not properly templated in the generated SQL.

Default value via query configuration:

```sql
[[AND {{report_date}} #]] AND report_date = DATE_SUB(CURRENT_DATE("America/Chicago"), INTERVAL 1 DAY)
```

Generated SQL:

```sql
AND 1 = 1 # AND report_date = DATE_SUB(CURRENT_DATE("America/Chicago"), INTERVAL 1 DAY)
```

Expected (desired) SQL:

```sql
AND report_date = DATE_SUB(CURRENT_DATE("America/Chicago"), INTERVAL 1 DAY)
```

**Comment by flamber** [comment]
This actually looks like a duplicate of #5541

**Comment by winnephew** [comment]
@camsaul: I think many of the date filter issues reported might be associated with #9222 . Basically I think "Database Default" might not be working properly or at least not the way I would expect it to.

My example:
I use mariaDB with timezone set to the default "system" which in my case currently is UTC+2. If I set the report timezone to "Database Default" I receive data that is offset by -1 day. If I explicitly set it to Berlin/Germany I instead receive the exact data that I would expect.

**Comment by camsaul** [comment]
Fixed by #10947
